README.md

```diff
@@ -690,3 +690,146 @@ In general, `for of` is recommended over `for in` or `forEach`
+<p>
+
+#### Answer: B
+
+The `array.unshift()` method in JavaScript adds one or more elements to the beginning of an array and returns the new length of the array. In the given code, arr is an array with the values `[3, 5, 7, 9]`. The `unshift()` method is called with the arguments 1 and 2, which adds 1 and 2 to the beginning of the arr array.
+
+The `unshift()` method returns the new length of the array after the addition of the element(s). In this case, the new length of arr will be 6 because 1 and 2 are added to the beginning of the array, shifting the existing elements to the right.
+
+</p>
+</details>
+
+---
+
+##### 21. Array Manipulation: Understanding array.pop() Method in JavaScript
+
+```javascript
+const myArray = [1, 2, 3, 4, 5];
+const poppedValue = myArray.pop();
+console.log(poppedValue);
+```
+
+What does the above code snippet output?
+
+- A: 1
+- B: 5
+- C: 'undefined'
+- D: An error will occur
+
+<details><summary><b>Answer</b></summary>
+<p>
+
+#### Answer: B
+
+The `Array.pop()` method in JavaScript removes the last element from an array and returns that element. In this case, myArray is an array with elements `[1, 2, 3, 4, 5]`, and `myArray.pop()` is called, which removes the element 5 from the array and returns it.
+
+</p>
+</details>
+
+---
+
+##### 22. Array Manipulation: Understanding array.shift() Method in JavaScript
+
+```javascript
+const arr = [10, 20, 30, 40, 50];
+const removedElement = arr.shift();
+console.log(removedElement);
+console.log(arr);
+```
+
+What does the above code snippet output?
+
+- A: 'removedElement' will be 10 and 'arr' will be [20, 30, 40, 50]
+- B: 'removedElement' will be 50 and 'arr' will be [10, 20, 30, 40]
+- C: 'removedElement' will be 10 and 'arr' will be [10, 20, 30, 40, 50]
+- D: 'removedElement' will be 50 and 'arr' will be [10, 20, 30, 40, 50]
+
+<details><summary><b>Answer</b></summary>
+<p>
+
+#### Answer: A
+
+The `Array.shift()` method removes the first element from the arr array, which is `10`, and returns it. The resulting arr array will then be `[20, 30, 40, 50]`. So, option A is the correct answer as it reflects the correct value of removedElement and the updated state of arr after `Array.shift()` is called. Option B is incorrect as it states that removedElement will be 50, which is not true. Option C and D are also incorrect as they state that arr remains unchanged, which is not accurate.
+
+</p>
+</details>
+
+---
+
+##### 23. Array Manipulation: Understanding array.splice() Method in JavaScript
+
+```javascript
+let arr = [1, 2, 3, 4, 5];
+let removed = arr.splice(1, 2, "a", "b");
+console.log(removed);
+console.log(arr);
+```
+
+What does the above code snippet output?
+
+- A: 'removed' will be [2, 3] and 'arr' will be [1, 'a', 'b', 4, 5]
+- B: 'removed' will be [2, 3] and 'arr' will be [1, 'a', 'b', 4, 5]
+- C: 'removed' will be [2, 3, 'a', 'b'] and 'arr' will be [1, 4, 5]
+- D: 'removed' will be [1, 2] and 'arr' will be [3, 4, 5]
```

Comment:
Please update A option. Since it is same as option B

```diff
- - A: 'removed' will be [2, 3] and 'arr' will be [1, 'a', 'b', 4, 5]
- - B: 'removed' will be [2, 3] and 'arr' will be [1, 'a', 'b', 4, 5]
- - C: 'removed' will be [2, 3, 'a', 'b'] and 'arr' will be [1, 4, 5]
- - D: 'removed' will be [1, 2] and 'arr' will be [3, 4, 5]
+ - A: removed: [1, 2] arr: [a, b, 3, 4, 5]
+ - B: removed: [2, 3] arr: [1, 'a', 'b', 4, 5]
+ - C: removed: [2, 3, 'a', 'b'] arr: [1, 4, 5]
+ - D: removed: [1, 2] arr: [3, 4, 5]
```